builder/azure/arm/step_deploy_template.go

```diff
@@ -299,9 +309,9 @@ func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, depl
 					resourceName,
 					resourceGroupName)
 				if err != nil {
-					s.say(fmt.Sprintf("Error deleting resource. Will retry.\n"+
```

Comment:
We regularly fail to delete attached network resources related to the resource being still attached, this leads to a lot of spam in the output, here I've changed this to only log the error when we fail for real, since almost every build I run fails this check at least once for at least two different resources.